File: Tarea6_1.cpp
#include<iostream>
using namespace std;

int main(){
    //Declaracion de productos y precios
    string producto[5]={"Leche", "Pan", "Yogurt", "Cereal", "Sandia"};
    float precio[5]={1,0.25,0.75,1.5,2.75};

    //Declaracion de variables a buscar
    string buscar;
    bool encontrado = false;

    //Pedir el producto al usuario
    cout << "Ingrese el producto que desea buscar: ";
    cin >> buscar;

    //Busqueda lineal con un for
    for(int i=0; i<5; i++){

        if(producto[i]==buscar){
            cout << "Producto encontrado!\n";
            cout << "Producto: " << producto[i] 
            <<"\nPrecio: " << precio[i] << endl;

            encontrado=true;
            break;
        }
    }

    //En caso de que no se encuentre el producto
    if(!encontrado){
        cout << "No se dispone del producto ingresado";
    }
    return 0;
}

File: Tarea6_1.java
import java.util.Scanner;

public class BuscarProducto {
    public static void main(String[] args) {
        // Declaración de un arreglo de productos
        String[] producto = {"Leche", "Pan", "Yogurt", "Cereal", "Sandia"};
        // Declaración de un arreglo de precios (correspondiente a cada producto)
        float[] precio = {1f, 0.25f, 0.75f, 1.5f, 2.75f};

        // Objeto Scanner para leer datos desde teclado
        Scanner sc = new Scanner(System.in);

        // Solicitar al usuario el producto que desea buscar
        System.out.print("Ingrese el producto que desea buscar: ");
        String buscar = sc.nextLine();
            
        // Variable booleana para indicar si se encontró el producto
        boolean encontrado = false;

        // Recorrer el arreglo de productos
        for (int i = 0; i < producto.length; i++) {
                // Comparar el producto ingresado con el arreglo (ignora mayúsculas/minúsculas)
            if (producto[i].equalsIgnoreCase(buscar)) {
                System.out.println("Producto encontrado!");
                System.out.println("Producto: " + producto[i]);
                System.out.println("Precio: " + precio[i]);
                encontrado = true; // marcar que sí se encontró
                break; // salir del ciclo porque ya se encontró
            }
        }

        // Si no se encontró el producto, mostrar mensaje
        if (!encontrado) {
            System.out.println("No se dispone del producto ingresado");
        }

        // Cerrar el objeto Scanner
        sc.close();
    }
}
